cylc/flow/config.py

```diff
@@ -271,12 +271,17 @@ def __init__(
             self.cfg['runtime']['root'] = OrderedDictWithDefaults()
 
         try:
-            parameter_values = self.cfg['scheduler']['parameters']
+            # Ugly hack to avoid templates getting included in parameters
+            parameter_values = {
+                key: value for key, value in
+                self.cfg['task parameters'].items()
+                if key != 'templates'
+            }
```

Comment:
Because `[templates]` is a subsection of `[task parameters]` we have to specifically remove them from the parameter values dict. 
There is a potential bug here: If a user were to have a template variable called "templates" (Not inconceivable?) then this would be removed leading to a hard to debug error. 
Question: Do we:
- Hope this problem never arises.
- Document this and forget about it.
- Add a warning if key == 'templates'.
- Prevent (raise error) if parameter_values includes templates that isn't a section. (I don't know if Parsec will do this cleverly (@hjoliver ?) but this is my favoured option).

Reply:
> Make the test at line 278 more specific using type of value as well as value of key.

Seems doable to me, something like:
```python
parameter_values = self.cfg['task parameters']
if 'templates' in parameter_values and isinstance(parameter_values['templates'], OrderedDict):
    parameter_values.pop('templates')
```
(Not tested)

Reply:
> Seems doable to me

Else I'd not have said it, interested in whether other people thought it a good idea....

Reply:
I'm not sure it's reasonable to have a template variable called "templates". It's reasonable to disallow it, because templates work just the same with different variable names (even "templatesx" or whatever if you insist).  However I don't mind to much if we handle it or exclude it.  Maybe handling it is best actually, just to avoid an exclusion that we have to document.